src/pcre2_compile.c

```diff
@@ -6078,17 +6101,19 @@ for (;; pptr++)
     case META_COND_RNUMBER:   /* (?(Rdigits) */
     case META_COND_NAME:      /* (?(name) or (?'name') or ?(<name>) */
     case META_COND_RNAME:     /* (?(R&name) - test for recursion */
-    case META_SCS_NAME:       /* (*scan_substring:'name') or (*scan_substring:(<name>)) */
-    case META_SCS_NEXT_NAME:  /* More names for scan substring. */
-    bravalue = meta == META_SCS_NAME ? OP_ASSERT_SCS : OP_COND;
+    case META_SCS_NAME:       /* Name of scan substring */
+    bravalue = OP_COND;
```

Comment:
Does it matter that we set `bravalue` to something a bit meaningless, for the `META_SCS_NAME` case?

It looks like `bravalue`is only used in `GROUP_PROCESS:`, and the code here for META_SCS_NAME has already done that, and every branch below which does the jump there checks for META_SCS_NAME to make sure it's special-cased.

Reply:
There are goto-s in the code, and this is something which is easy to forget adding them before every goto (also increase code size). I would not change it, even if it is a dead code in the scs case.